governance/xc_admin/packages/xc_admin_common/src/governance_payload/SetDataSources.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  ActionName,
+  PythGovernanceAction,
+  PythGovernanceActionImpl,
+  PythGovernanceHeader,
+} from "./PythGovernanceAction";
+import { ChainName } from "@certusone/wormhole-sdk";
+import * as BufferLayout from "@solana/buffer-layout";
+import * as BufferLayoutExt from "./BufferLayoutExt";
+
+/** A data source is a wormhole emitter, i.e., a specific contract on a specific chain. */
+export interface DataSource {
+  emitterChain: number;
+  emitterAddress: string;
+}
+const DataSourceLayout: BufferLayout.Structure<DataSource> =
+  BufferLayout.struct([
+    BufferLayout.u16("emitterChain"),
+    BufferLayoutExt.hexBytes(32, "emitterAddress"),
+  ]);
```

Comment:
I think this layout is wrong, there's a u8 first that contains the length of the vector of DataSources